airflow/ti_deps/deps/trigger_rule_dep.py

```diff
@@ -213,13 +230,18 @@ def _iter_upstream_conditions() -> Iterator[ColumnOperators]:
         # "simple" tasks (no task or task group mapping involved).
         if not any(needs_expansion(t) for t in upstream_tasks.values()):
             upstream = len(upstream_tasks)
+            upstream_setup = len(
+                [x for x in upstream_tasks.values() if not isinstance(x, MappedOperator) and x._is_setup]
+            )
         else:
             upstream = (
                 session.query(func.count())
                 .filter(TaskInstance.dag_id == ti.dag_id, TaskInstance.run_id == ti.run_id)
                 .filter(or_(*_iter_upstream_conditions()))
                 .scalar()
             )
+            # todo: add support for mapped setup?
```

Comment:
Can you elaborate on this one?
What is the issue with mapped tasks?

Reply:
Just complexity that I am inclined to deferr for the moment.  Not sure we really need them.
To calculate upstream for mapped tasks, we determine upstream count by querying the database.  In order to be able to do that for setup tasks, we'd need to have setup tasks marked as such in the database, or use some sort of hybrid logic.